lib/fdk/runner.rb

```diff
@@ -5,121 +5,45 @@
-                        }
-                        STDOUT.puts response.to_json
-                        STDOUT.puts
-                        STDOUT.flush
-                        first_bracket = false
-                        obs = ""
-                    end
-                end
-            end
-
-            # STDIN.each do |line|
-            #     STDERR.puts "LINE: --#{line}--"
-            #     STDERR.flush
-            #     ls = line.strip
-            #     STDERR.puts "ls: #{ls}"
-            #     if ls == "}"
-            #         # STDERR.puts "endbracket true"
-            #         endbracket=true
-            #     elsif ls == "" && endbracket
-            #         # TODO: this isn't very robust, probably needs to be better :/
-            #         STDERR.puts "OBJECT: #{obs}"
-            #         payload = JSON.parse(obs)
-            #         STDERR.puts "payload: #{payload.inspect}"
-            #         c = Context.new(payload)
-            #         STDERR.puts "context: " + c.inspect
-            #         STDERR.flush
-            #         body = payload['body']
-            #         if c.content_type == 'application/json'
-            #             body = JSON.parse(body)
-            #         end
-            #         # TODO: begin/rescue so we can respond with proper error response and code
-            #         s = FDK.single_event(func, c, body)
-            #         response = {
-            #             headers: {
-            #                 'Content-Type' => 'application/json'
-            #             },
-            #             'status_code' => 200,
-            #             body: s.to_json,
-            #         }
-            #         STDOUT.puts response.to_json
-            #         STDOUT.puts
-            #         STDOUT.flush
-            #         obs = ""
-            #         next
-            #     else 
-            #         endbracket = false
-            #     end
-            #     obs += line
-            # end
-        elsif format == "default"
-            # TODO: check if content type json, and if so, parse it before passing it in
-            body = STDIN.read
-            payload = {}
-            payload['call_id'] = ENV['FN_CALL_ID']
-            payload['content_type'] = ENV['FN_HEADER_Content_Type']
-            payload['protocol'] = {
-                'type' => 'http',
-                'request_url' => ENV['FN_REQUEST_URL']
-            }
-            # STDERR.puts "payload: #{payload}"
-            c = Context.new(payload)
-            if c.content_type == "application/json"
-                # STDERR.puts "parsing json"
-                body = JSON.parse(body)
-            end
-            puts FDK.single_event(func, c, body).to_json
-        else
-            raise "Format '#{format}' not supported in Ruby FDK."
-        end
+  def self.handle(func)
+    format = ENV['FN_FORMAT']
+    if format == 'json'
+      payload = JSON.parse(STDIN.read)
```

Comment:
I'm pretty sure this doesn't work, it's not going to be able to handle multiple requests.